fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadJob.java

```diff
@@ -1056,7 +1062,11 @@ public void replayOnAborted(TransactionState txnState) {
             failMsg = new FailMsg(FailMsg.CancelType.LOAD_RUN_FAIL, txnState.getReason());
             finishTimestamp = txnState.getFinishTime();
             state = JobState.CANCELLED;
-            GlobalStateMgr.getCurrentState().getGlobalTransactionMgr().getCallbackFactory().removeCallback(id);
+            if (retryTime <= 0 || !failMsg.getMsg().contains("timeout") || !isTimeout()) {
```

Comment:
string contains as the check is not stable enough?

Reply:
Also with `isTimeout()`